core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -296,6 +332,9 @@ private[spark] class Executor(
       } else {
         taskRunner.kill(interruptThread = interruptThread, reason = reason)
       }
+    } else {
+      // Safe to remove kill mark as we got a chance with the TaskRunner.
```

Comment:
this else branch means `taskRunner == null`?

Reply:
oh... good catch